integration_tests/ctesque/src/sharedTest/java/android/app/UiAutomationTest.java

```diff
@@ -55,4 +66,23 @@ public void setRotation_freeze180_isPortrait() {
           });
     }
   }
+
+  private static void waitDisplayRotation(
+      Activity activity, int expectedRotation, long timeOutMilliSeconds) {
+    long startMs = SystemClock.uptimeMillis();
+    long currentMs = SystemClock.uptimeMillis();
+    while (currentMs - startMs <= timeOutMilliSeconds) {
+      Display display = activity.getWindowManager().getDefaultDisplay();
+      if (display.getRotation() == expectedRotation) {
+        break;
+      }
+      try {
+        // Sleep 100ms to avoid unnecessary checking.
+        Thread.sleep(100);
+      } catch (InterruptedException e) {
+        // Do nothing
+      }
+      currentMs = SystemClock.uptimeMillis();
+    }
```

Comment:
```suggestion
    do {
      Display display = activity.getWindowManager().getDefaultDisplay();
      if (display.getRotation() == expectedRotation) {
        break;
      }
      try {
        // Sleep 100ms to avoid unnecessary checking.
        Thread.sleep(100);
      } catch (InterruptedException e) {
        // Do nothing
      }
    } while (SystemClock.uptimeMillis() - startMs <= timeOutMilliSeconds);
```